ext/OceananigansXESMFExt.jl

```diff
@@ -0,0 +1,159 @@
+module OceananigansXESMFExt
+
+using XESMF
+using CUDA
+using SparseArrays
+using Oceananigans
+using Oceananigans.Architectures: CPU, architecture
+using Oceananigans.Grids: λnodes, φnodes, Center, Face
+
+import Oceananigans.Architectures: on_architecture
+import Oceananigans.Fields: Regridder
+
+
+function x_node_array(x::AbstractVector, Nx, Ny)
+    return Array(repeat(view(x, 1:Nx), 1, Ny))'
+end
+function  y_node_array(x::AbstractVector, Nx, Ny)
+    return Array(repeat(view(x, 1:Ny)', Nx, 1))'
+end
+x_node_array(x::AbstractMatrix, Nx, Ny) = Array(view(x, 1:Nx, 1:Ny))'
+
+function x_vertex_array(x::AbstractVector, Nx, Ny)
+    return Array(repeat(view(x, 1:Nx+1), 1, Ny+1))'
+end
+function y_vertex_array(x::AbstractVector, Nx, Ny)
+    return Array(repeat(view(x, 1:Ny+1)', Nx+1, 1))'
+end
+x_vertex_array(x::AbstractMatrix, Nx, Ny) = Array(view(x, 1:Nx+1, 1:Ny+1))'
+
+y_node_array(x::AbstractMatrix, Nx, Ny) = x_node_array(x, Nx, Ny)
+y_vertex_array(x::AbstractMatrix, Nx, Ny) = x_vertex_array(x, Nx, Ny)
+
+function extract_xesmf_coordinates_structure(dst_field, src_field)
```

Comment:
I think should be a general function of the `XESMF.jl` package, imported here and extended for the case of 

```suggestion
function extract_xesmf_coordinates_structure(dst_field::AbstractField, src_field::AbstractField)
```